src/FluentResults/Results/Result.cs

```diff
@@ -176,15 +170,24 @@ public async ValueTask<Result> Bind(Func<ValueTask<Result>> action)
         {
             var result = new Result();
             result.WithReasons(Reasons);
-            
-            if (IsSuccess)
-            {
+
+            if (IsSuccess) {
                 var converted = await action();
                 result.WithReasons(converted.Reasons);
             }
 
             return result;
         }
+
+        public static implicit operator Result(Error error)
```

Comment:
Please add some tests for these implicit conversations for Result and Result of T